jsoc/gsoc/juliaconstraints.md

```diff
@@ -0,0 +1,54 @@
+# Constraint Programming in Julia
+
+[JuliaConstraints](https://github.com/JuliaConstraints) is an organization supporting packages for Constraint Programming in Julia. Although it is independent of it, it aims for a tight integration with JuMP.jl over time. For a detailed overview of basic Constraint Programming in Julia, please check our video from JuliaCon 2021 [Put some constraints into your life with JuliaCon(straints)](https://youtu.be/G4siuvNMj0c).
+
+### General goal of JuliaConstraints
+
+Often, problem-solving involves taking two actions: model and solve. Typically, there is a trade-off between ease of modeling and efficiency of solving. Therefore, one is often required to be a specialist to model and solve an optimization problem efficiently. We investigate the theoretical fundamentals and the implementation of tools to automize and make optimization frameworks. A general user should focus on the model of practical problems, regardless of the software or hardware available. Furthermore, we aim to encourage technical users to use our tools to improve their solving efficiency.
+
+**Mentor:** [Jean-Francois Baffier](http://baffier.fr/) ([azzaare@github](https://github.com/Azzaare))
+
+## [PerfChecker.jl](https://github.com/JuliaConstraints/PerfChecker.jl)
+
+This package consists of a set of tools designed to check the performance of packages over time and versions. The targeted audience is the whole community of packages' developers in Julia (not only JuliaConstraints).
+
+The [README](https://github.com/JuliaConstraints/PerfChecker.jl) provides a short *demo* on how PerfChecker can be used.
+
+A non-exhaustive list of wanted features:
+
+    - PerfCheck environment similar to Test.jl and Pkg.jl
+    - Sugar syntax `@bench`, `@alloc`, `@profile` similar to Test.jl and Pkg.jl
+    - Automatic plotting of previous features
+    - Interactive REPL interface
+    - Interactive GUI interface (using for instance Makie)
+    - Automatic Profiling ? (not sure, there already is a bunch of super cool packages)
+    - Smart semi-automatic analysis of performance through Machine Learning
```

Comment:
Generally, the GSoC staff (and students) are looking to have specific things detailed instead of a list of possible features. Can you turn these into maybe two different groups and then say that features are interchangeable depending on the interest of the candidate?

Reply:
That is a nice suggestion, thanks.